### PR TITLE
【Comm】switch c_allreduce_* to all_reduce

### PR DESCRIPTION
**Pull request by Xing-lil**
### PR Category

Communication Library

### PR Types

Improvements

### Description

Pcard-70448

### CODE REVIEW

File: paddle/fluid/framework/ir/graph_helper.cc [hunk 1]
```diff
@@ -525,12 +525,12 @@ void ReplaceAllReduceOp(const Node &node,
     all_reduce_var_name = in_var_handles[0]->Name();
   }
 
-  // add c_allreduce_sum OP
+  // add all_reduce OP
```
Comment (liym27) [inline paddle/fluid/framework/ir/graph_helper.cc]: -> // add all_reduce(sum) OP

File: paddle/fluid/framework/ir/pass_tester_helper.h [hunk 2]
```diff
@@ -184,12 +184,11 @@ struct Layers {
     return unary_op("c_identity", x, out, &attrs);
   }
 
-  VarDesc* c_allreduce_sum(VarDesc* x,
-                           VarDesc* out = nullptr,
-                           int ring_id = -1) {
+  VarDesc* all_reduce(VarDesc* x, VarDesc* out = nullptr, int ring_id = -1) {
```
Comment (liym27) [inline paddle/fluid/framework/ir/pass_tester_helper.h]: all_reduce  ->  all_reduce_sum, and modify all calls in `fused_multi_transformer_decoder_pass_tester.cc`~

File: paddle/fluid/framework/ir/fused_multi_transformer_decoder_pass_tester.cc [hunk 3]
```diff
@@ -474,9 +474,9 @@ TEST(MultiDevicesFusedMultiTransformerDecoderFuseQKVPass, basic) {
   auto* bias_l = layers.data("weightsl", {1024, 1024}, true);
   auto* linear_matmut_out =
       layers.matmul_v2(reshape_qkv_out, weights_l, nullptr, false, true);
-  auto* c_allreduce_out = layers.c_allreduce_sum(linear_matmut_out);
+  auto* all_reduce_out = layers.all_reduce(linear_matmut_out);
```
Comment (liym27) [inline paddle/fluid/framework/ir/fused_multi_transformer_decoder_pass_tester.cc]: all_reduce_out -> allreduce_sum_out

File: paddle/fluid/framework/ir/fused_multi_transformer_encoder_pass.h [hunk 4]
```diff
@@ -301,8 +301,8 @@ struct MultiDevicesFusedMultiTransformerEncoderPattern : public PatternBase {
   PATTERN_DECL_NODE(matmul_linear);
   PATTERN_DECL_NODE(matmul_linear_w);
   PATTERN_DECL_NODE(matmul_linear_out);
-  PATTERN_DECL_NODE(c_allreduce_sum);
-  PATTERN_DECL_NODE(c_allreduce_sum_out);
+  PATTERN_DECL_NODE(all_reduce);
```
Comment (liym27) [inline paddle/fluid/framework/ir/fused_multi_transformer_encoder_pass.h]: all_reduce -> all_reduce_sum

File: paddle/fluid/framework/ir/graph_helper.cc (hunk 1 above)
Comment (Xing-lil) [reply paddle/fluid/framework/ir/graph_helper.cc]: Thx, the code has been modified.

File: paddle/fluid/framework/ir/pass_tester_helper.h (hunk 2 above)
Comment (Xing-lil) [reply paddle/fluid/framework/ir/pass_tester_helper.h]: Thx, the variable names has been modified.

File: paddle/fluid/framework/ir/fused_multi_transformer_decoder_pass_tester.cc (hunk 3 above)
Comment (Xing-lil) [reply paddle/fluid/framework/ir/fused_multi_transformer_decoder_pass_tester.cc]: Thx, the variable names has been modified.

File: paddle/fluid/framework/ir/fused_multi_transformer_encoder_pass.h (hunk 4 above)
Comment (Xing-lil) [reply paddle/fluid/framework/ir/fused_multi_transformer_encoder_pass.h]: Thx, the code has been modified.

**Review by liym27 — APPROVED** [review]

**Review by liym27 — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Review by zhangbo9674 — APPROVED** [review]
LGTM

**Pull request merged**